src/librustdoc/doctree.rs

```diff
@@ -4,17 +4,48 @@ use rustc_span::{self, Span, Symbol};
 
 use rustc_hir as hir;
 
+/// A warp around an hir::Item
+#[derive(Debug)]
+pub(crate) struct Item<'hir> {
+    /// the wrapped item
+    pub(crate) hir_item: &'hir hir::Item<'hir>,
+    /// the explicit renamed name
+    pub(crate) renamed_name: Option<Symbol>,
+    /// whether the item is from a glob import
+    /// if `from_glob` is true and we see another item with same name,
+    /// then this item can be replaced with that one
+    pub(crate) from_glob: bool,
+}
```

Comment:
nit: I would use `crate` to be consistent with the rest of rustdoc.
```suggestion
crate struct Item<'hir> {
    /// the wrapped item
    crate hir_item: &'hir hir::Item<'hir>,
    /// the explicit renamed name
    crate renamed_name: Option<Symbol>,
    /// whether the item is from a glob import
    /// if `from_glob` is true and we see another item with same name,
    /// then this item can be replaced with that one
    crate from_glob: bool,
}
```